syne_tune/optimizer/schedulers/searchers/model_based_searcher.py

```diff
@@ -487,3 +479,331 @@ def _assign_random_searcher(self):
                 restrict_configurations=self._restrict_configurations,
             )
             self._random_searcher.set_random_state(self.random_state)
+
+
+def create_initial_candidates_scorer(
+    initial_scoring: str,
+    model: SurrogateOutputModel,
+    acquisition_class: AcquisitionClassAndArgs,
+    random_state: np.random.RandomState,
+    active_metric: str = INTERNAL_METRIC_NAME,
+) -> ScoringFunction:
+    if initial_scoring == "thompson_indep":
+        if isinstance(model, dict):
+            assert active_metric in model
+            model = model[active_metric]
+        return IndependentThompsonSampling(model, random_state=random_state)
+    else:
+        acquisition_class, acquisition_kwargs = unwrap_acquisition_class_and_kwargs(
+            acquisition_class
+        )
+        return acquisition_class(
+            model, active_metric=active_metric, **acquisition_kwargs
+        )
+
+
+class BayesianOptimizationSearcher(ModelBasedSearcher):
```

Comment:
Its is kind of hard to compare line-by-line. Could you summarize what changed between this implementation and the one we have in GPFIFOSeacher? It seems that its mostly the same but I want to make sure I havent missed anything.

Reply:
All code was moved to `BayesianOptimizationSearcher`, except for the constructor and the method to restore from inner state.

Construction of our complex searchers is always done by a factory, because there is a lot of code (and complexity) which is shared.

If you come up with a new one, we also need to make sure the construction is easy for the user.